### PR TITLE
add json of row groupings - mock to fill out later

### PR DESCRIPTION
**Pull request by PeterDSteinberg**
* <strike>[x] Adds a json of row labels to group names: `btax/param_defaults/row_groupings.json`
* [ ] We need to add the correct group labels and `summary_c` and `summary_nc` to the row labels json 
* [x] Access the row groupings with:
```python
from btax.parameters import DEFAULT_ROW_GROUPINGS`
```
</strike>

Edit: This PR now takes the asset and industry categories from the dataframes returned by `run_btax` (avoids adding json to repo).
This PR works in conjunction with [webapp-public PR 377](https://github.com/OpenSourcePolicyCenter/webapp-public/pull/377)

### CODE REVIEW

**Comment by PeterDSteinberg** [comment]
@jdebacker Can you fill out real content in the `param_defaults/row_groupings.json` and make PR against this branch?

**Comment by jdebacker** [comment]
@PeterDSteinberg If we want to order the items in the groups, can we (and should we) add that information here?  Let me know and I'll do it all at once.  I would like to specify the order the groups are presented in as well as the items in the group.

**Comment by PeterDSteinberg** [comment]
@jdebacker The current dict of dicts format won't preserve order.  Let me push a change to the structure that will allow us to preserve the order in which they are written in the json.

**Comment by talumbau** [comment]
@PeterDSteinberg does the commit by @jdebacker resolve the last checkpoint on this PR?

**Comment by jdebacker** [comment]
@PeterDSteinberg in the `row_groupings.json` file, there are assets and industries listed.  How am I differentiating between the two?  Or do I?  For example, will the there be an industry with `"group": "grouping_number_2"` and an asset with `"group": "grouping_number_2"`?

**Comment by jdebacker** [comment]
@PeterDSteinberg, also, I'm assuming the group number identifies the group the item is in and the order in which the items are listed will defined the order within the groups, is that correct?

**Comment by PeterDSteinberg** [comment]
@jdebacker There is a separate list for "industry" ([here](https://github.com/open-source-economics/B-Tax/pull/113/files#diff-b1370b15028275b110a09228eef62ae3R2)) and "asset" ([here](https://github.com/open-source-economics/B-Tax/pull/113/files#diff-b1370b15028275b110a09228eef62ae3R604))

The ordering within that list determines the order of the rows and the "group" becomes the grouping name in the UI.  I am not sure what `summary_nc` and `summary_c` are, but they will show up at the top row of each grouping as summaries.

@talumbau we still need to commit real content to this PR before merging.  It still has dummy information of the right structure.

**Comment by jdebacker** [comment]
@PeterDSteinberg Looking at web-app [PR #377](https://github.com/OpenSourcePolicyCenter/webapp-public/pull/377), I think what we need instead of the `summary_nc` and `summary_c` values are just values for each of the output variables for the group.  Can we do that?

In PR #110 , I include these groups are separate rows in the data frames output from B-Tax, with all the variables calculated at the group level.  Note that I also include a column so one can match each item (asset or industry) with the group it is a part of - I kind of thought that information would allow one to avoid a json file like we are trying to build here).  

Let me know if:
1) We can include the output variables at the group-level (which are now computed in B-Tax).
2) We can use the new data frames (with the additional column identifying the group that each row of the table belongs to) to help simplify/avoid the json file proposed here.

**Comment by PeterDSteinberg** [comment]
@jdebacker I didn't realize we already essentially had this info in the source.  Can you point out a code snippet of exactly where I should look for the dataframes with the grouping info and I can work on revising this PR to use that info, removing the json added here.  

I think we should make a separate issue to do the change you mentioned:
```
@PeterDSteinberg Looking at web-app PR #377, I think what we need instead of the summary_nc and summary_c values are just values for each of the output variables for the group. Can we do that?
```
It would require a change in the front end assets @brendancol  has developed and I don't think I could get it done today for deploy.  We can make a separate issue for it.  I think the UI for deploy today is hard coded to using the `summary_c` and `summary_nc` fields (see below), so we should come up with temporary plan for using them.  That's just my thoughts on the order of operations - is that okay @talumbau @jdebacker ?

screen shot 2016-10-25 at 8 55 15 am

**Comment by jdebacker** [comment]
@PeterDSteinberg The data frames returned from `run_btax`  (i.e., [here](https://github.com/open-source-economics/B-Tax/blob/master/btax/run_btax.py#L90), now contains columns for the major group.  For the `output_by_asset` table, this column is named `major_asset_group`. In the `output_by_industry` table, this column is named `major_industry`.  

You'll also notice that for the average over group, the item identifier = the group name (e.g., the average of the METR and other variables for all type of equipment are listed in the `output_by_asset` data frame in a row where `Asset Type` =  `major asset_group` = `Equipment`).

File: btax/run_btax.py [hunk 1]
```diff
@@ -93,6 +93,22 @@ def run_btax(test_run,baseline=False,start_year=2016,iit_reform=None,**user_para
 def run_btax_with_baseline_delta(test_run,start_year,iit_reform,**user_params):
     econ_params = filter_user_params_for_econ(**user_params)
     base_output_by_asset, base_output_by_industry = run_btax(test_run,True,start_year,{},**econ_params)
+    asset_row_grouping = {}
+    subset = zip(*(getattr(base_output_by_asset, at) for at in ('Asset', 'asset_category', 'mettr_c', 'mettr_nc')))
+    for asset, cat, mettr_c, mettr_nc in subset:
+        if cat != cat:  # A string column that may have NaN, so can't do isnan()
+            cat = asset # These are some summary rows that don't have all info
```
Comment (PeterDSteinberg) [inline btax/run_btax.py]: @jdebacker Is this special case for NaN `asset_category` concerning?  When I ran it locally I found these were the rows of `base_output_by_asset` that had NaN `asset_category`:

```
In [35]: base_output_by_asset[base_output_by_asset.asset_category.isnull()]
Out[35]:
    ADS            Asset       Asset Type  Class Life  GDS Method System  \
76  NaN        Equipment        Equipment         NaN  NaN    NaN    NaN
77  NaN       Structures       Structures         NaN  NaN    NaN    NaN
78  NaN  All Investments  All Investments         NaN  NaN    NaN    NaN

   asset_category      assets_c     assets_nc    ...      rho_c_e    rho_nc  \
76            NaN  4.770213e+12  1.673524e+12    ...     0.068006  0.049366
77            NaN  8.826705e+12  4.852032e+12    ...     0.077886  0.057511
78            NaN  1.872397e+13  1.651470e+13    ...     0.078951  0.059155

    rho_nc_d  rho_nc_e       z_c     z_c_d     z_c_e      z_nc    z_nc_d  \
76  0.028211  0.058064  0.912424  0.941076  0.900115  0.922922  0.942649
77  0.032631  0.067830  0.592910  0.682486  0.562458  0.479208  0.563926
78  0.033370  0.069738  0.511958  0.561485  0.494467  0.234316  0.261206

      z_nc_e
76  0.915298
77  0.452099
78  0.225579
```

I guessed defaulting to the Asset or Asset_Type was ok.

**Comment by jdebacker** [comment]
Thanks @PeterDSteinberg, merging now.

**Pull request merged**